src/apps/KintoAppRegistry.sol

```diff
@@ -255,19 +337,60 @@ contract KintoAppRegistry is
     }
 
     /**
-     * @dev Allows the owner to override the parent contract of a child contract
-     * @param child The address of the child contract
-     * @param parent The address of the parent contract
+     * @notice Determines if a contract call is allowed from an EOA (Externally Owned Account)
+     * @dev This function checks various conditions to decide if an EOA can call a specific contract:
+     *      1. Allows calls to system contracts from any EOA
+     *      2. Checks if the EOA has a linked wallet
+     *      3. Verifies if dev mode is enabled on the wallet
+     *      4. Ensures the wallet owner has completed KYC
+     *      5. Permits CREATE and CREATE2 operations for eligible EOAs
+     *      6. Allows dev EOAs to call their respective apps
+     * @param from The address of the EOA initiating the call
+     * @param to The address of the contract being called
+     * @return A boolean indicating whether the contract call is allowed (true) or not (false)
      */
-    function overrideChildToParentContract(address child, address parent) external override onlyOwner {
-        childToParentContract[child] = parent;
+    function isContractCallAllowedFromEOA(address from, address to) external view returns (bool) {
+        // Calls to system contracts are allwed for any EOA
+        if (isSystemContract[to]) return true;
+
+        // Deployer EOAs are allowed to use CREATE and CREATE2
+        if (to == address(0) || to == CREATE2) {
```

Comment:
what happens when burning tokens? to is also address(0) right?

Reply:
This is unrelated to ERC20. The zero address is a parameter for the transfer method. This function is for the transaction object of the Ethereum chain, which is a different concept. In other words, this function is never invoked for ERC20.transfer because ERC20.transfer has FROM == USER and TO == ERC20Token.